src/exchange/providers/anthropic.py

```diff
@@ -138,26 +138,14 @@ def complete(
         )
         payload = {k: v for k, v in payload.items() if v}
 
-        max_retries = 5
-        initial_wait = 10  # Start with 10 seconds
-        backoff_factor = 1
-        for retry in range(max_retries):
-            response = self.client.post(ANTHROPIC_HOST, json=payload)
-            if response.status_code not in (429, 529, 500):
-                break
-            else:
-                sleep_time = initial_wait + (backoff_factor * (2**retry))
-                time.sleep(sleep_time)
-
-        if response.status_code in (429, 529, 500):
-            raise httpx.HTTPStatusError(
-                f"Failed after {max_retries} retries due to rate limiting",
-                request=response.request,
-                response=response,
-            )
+        response = self._send_request(payload)
 
         response_data = raise_for_status(response).json()
         message = self.anthropic_response_to_message(response_data)
         usage = self.get_usage(response_data)
 
         return message, usage
+    
+    @retry_httpx_request()
+    def _send_request(self, payload: Dict[str, Any]) -> httpx.Response:
```

Comment:
Should we figure out if we can allow the calling app (e.g. goose) to pass in options to configure the retry behavior?

Reply:
I think we can omit that as part of this PR.